pyproject.toml

```
@@ -0,0 +1,181 @@
+jax = [
+    "jax>=0.4,<=0.5",
+    "jaxlib>=0.4,<=0.5",
+]
+# For the scikits.odes solver
+odes = [
+    "scikits.odes"
+]
+# Contains all optional dependencies, except for odes, jax, and dev dependencies
+all = [
+    "anytree>=2.4.3",
+    "autograd>=1.2",
+    "pandas>=0.24",
+    "scikit-fem>=0.2.0",
+    "imageio>=2.9.0",
+    "matplotlib>=2.0",
+    "pybtex>=0.24.0",
+    "sympy>=1.8",
+    "bpx",
+    "tqdm",
+    "jupyter",
+]
+
+[project.scripts]
+pybamm_edit_parameter = "pybamm.parameters_cli:edit_parameter"
+pybamm_add_parameter = "pybamm.parameters_cli:add_parameter"
+pybamm_rm_parameter = "pybamm.parameters_cli:remove_parameter"
+pybamm_install_odes = "pybamm.install_odes:main"
+pybamm_install_jax = "pybamm.util:install_jax"
+
+[project.entry-points."pybamm_parameter_sets"]
+Sulzer2019 = "pybamm.input.parameters.lead_acid.Sulzer2019:get_parameter_values"
+Ai2020 = "pybamm.input.parameters.lithium_ion.Ai2020:get_parameter_values"
+Chen2020 = "pybamm.input.parameters.lithium_ion.Chen2020:get_parameter_values"
+Chen2020_composite = "pybamm.input.parameters.lithium_ion.Chen2020_composite:get_parameter_values"
+Ecker2015 = "pybamm.input.parameters.lithium_ion.Ecker2015:get_parameter_values"
+Ecker2015_graphite_halfcell = "pybamm.input.parameters.lithium_ion.Ecker2015_graphite_halfcell:get_parameter_values"
+Marquis2019 = "pybamm.input.parameters.lithium_ion.Marquis2019:get_parameter_values"
+Mohtat2020 = "pybamm.input.parameters.lithium_ion.Mohtat2020:get_parameter_values"
+NCA_Kim2011 = "pybamm.input.parameters.lithium_ion.NCA_Kim2011:get_parameter_values"
+OKane2022 = "pybamm.input.parameters.lithium_ion.OKane2022:get_parameter_values"
+OKane2022_graphite_SiOx_halfcell = "pybamm.input.parameters.lithium_ion.OKane2022_graphite_SiOx_halfcell:get_parameter_values"
+ORegan2022 = "pybamm.input.parameters.lithium_ion.ORegan2022:get_parameter_values"
+Prada2013 = "pybamm.input.parameters.lithium_ion.Prada2013:get_parameter_values"
+Ramadass2004 = "pybamm.input.parameters.lithium_ion.Ramadass2004:get_parameter_values"
+Xu2019 = "pybamm.input.parameters.lithium_ion.Xu2019:get_parameter_values"
+ECM_Example = "pybamm.input.parameters.ecm.example_set:get_parameter_values"
+MSMR_Example = "pybamm.input.parameters.lithium_ion.MSMR_example_set:get_parameter_values"
+
+[tool.setuptools]
+include-package-data = true
+
+# List of files to include as package data. These are mainly the parameter CSV files in
+# the input/parameters/ subdirectories. Other files such as the CITATIONS file, relevant
+# README.md files, and specific .txt files inside the pybamm/ directory are also included.
+# These are specified to be included in the SDist through MANIFEST.in.
+[tool.setuptools.package-data]
+pybamm = [
+    "*.txt",
+    "*.md",
+    "*.csv",
+    "*.py",
+    "pybamm/CITATIONS.bib",
+    "pybamm/plotting/mplstyle",
+]
+
+[tool.setuptools.packages.find]
+include = ["pybamm", "pybamm.*"]
```

See version comment above
```suggestion
[tool.setuptools.packages.find]
include = ["pybamm", "pybamm.*"]

[tool.setuptools.dynamic]
version = {attr = "pybamm.__version__"}
```